src/dscanner/analysis/redundant_storage_class.d

```diff
@@ -5,68 +5,106 @@
 
 module dscanner.analysis.redundant_storage_class;
 
-import std.stdio;
-import std.string;
-import dparse.ast;
-import dparse.lexer;
 import dscanner.analysis.base;
 import dscanner.analysis.helpers;
-import dsymbol.scope_ : Scope;
+import std.format;
+import std.algorithm.comparison : among;
+import std.algorithm.searching: all;
+import dmd.astenums : STC;
 
 /**
  * Checks for redundant storage classes such immutable and __gshared, static and __gshared
  */
-final class RedundantStorageClassCheck : BaseAnalyzer
+extern(C++) class RedundantStorageClassCheck(AST) : BaseAnalyzerDmd
 {
-	alias visit = BaseAnalyzer.visit;
-	enum string REDUNDANT_VARIABLE_ATTRIBUTES = "Variable declaration for `%s` has redundant attributes (%-(`%s`%|, %)).";
 	mixin AnalyzerInfo!"redundant_storage_classes";
+	alias visit = BaseAnalyzerDmd.visit;
 
-	this(string fileName, bool skipTests = false)
+	extern(D) this(string fileName, bool skipTests = false)
 	{
-		super(fileName, null, skipTests);
+		super(fileName, skipTests);
 	}
 
-	override void visit(const Declaration node)
+	override void visit(AST.VarDeclaration d)
 	{
-		checkAttributes(node);
-		node.accept(this);
-	}
+		const(char[])[] stcAttributes = [];
 
-	void checkAttributes(const Declaration node)
```

Comment:
This is the method used to perform the check, and here it's checked that the `Declaration` is in fact a `VariableDeclaration`, so the method `checkVariableDeclaration` could be called